make a way to disable callgrind

EscapeZero opened the pull request:
Summary: Ideally I would just use one of the existing preprocessor flags such as `FBCODE_CAFFE2`, but this implies a whole bunch of other things elsewhere, so it is not really a solution for ovrsource.

Test Plan: CI green, we are able to disable it internally with `-DNVALGRIND`

Differential Revision: D24227360

robieta reviewed (approved):
LGTM. Thanks for the fix.

malfet reviewed (changes requested):
Thank you for the patch, but can you please extend it a bit and add a corresponding CMAKE option?

EscapeZero committed:
make a way to disable callgrind (#46116)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/46116

Ideally I would just use one of the existing preprocessor flags such as `FBCODE_CAFFE2`, but this implies a whole bunch of other things elsewhere, so it is not really a solution for ovrsource.

Test Plan: CI green, we are able to disable it internally with `-DNVALGRIND`

Differential Revision: D24227360

fbshipit-source-id: 2145827bbb1fad22f4885b0c0a64352732c0de14
